--- a/stdenv/generic/check-meta.nix
+++ b/stdenv/generic/check-meta.nix
@@ -56,7 +56,7 @@
   # messages and make problems easier to spot.
   inHydra = config.inHydra or false;
   # Allow the user to opt-into additional warnings, e.g.
-  # import <corepkgs> { config = { showDerivationWarnings = [ "has-maintainers" ]; }; }
+  # import <nixpkgs> { config = { showDerivationWarnings = [ "maintainerless" ]; }; }
   showWarnings = config.showDerivationWarnings;

   getNameWithVersion =
@@ -118,8 +118,20 @@

   hasUnfreeLicense = attrs: attrs ? meta.license && isUnfree attrs.meta.license;

-  hasMaintainersOrTeams =
-    attrs: (attrs.meta.maintainers or [ ] != [ ]) || (attrs.meta.teams or [ ] != [ ]);
+  hasNoMaintainers =
+    # To get usable output, we want to avoid flagging "internal" derivations.
+    # Because we do not have a way to reliably decide between internal or
+    # external derivation, some heuristics are required to decide.
+    #
+    # If `outputHash` is defined, the derivation is a FOD, such as the output of a fetcher.
+    # If `description` is not defined, the derivation is probably not a package.
+    # Simply checking whether `meta` is defined is insufficient,
+    # as some fetchers and trivial builders do define meta.
+    attrs:
+    (!attrs ? outputHash)
+    && (attrs ? meta.description)
+    && (attrs.meta.maintainers or [ ] == [ ])
+    && (attrs.meta.teams or [ ] == [ ]);

   isMarkedBroken = attrs: attrs.meta.broken or false;

@@ -507,13 +519,13 @@
     else
       null;

-  # Please also update the type in /config/package-options.nix alongside this.
+  # Please also update the type in /pkgs/top-level/config.nix alongside this.
   checkWarnings =
     attrs:
-    if hasMaintainersOrTeams attrs then
-      {
-        reason = "has-maintainers";
-        errormsg = "has maintainers or teams defined in meta";
+    if hasNoMaintainers attrs then
+      {
+        reason = "maintainerless";
+        errormsg = "has no maintainers or teams";
         remediation = "";
       }
     else
