--- a/stdenv/impure.nix
+++ b/stdenv/impure.nix
@@ -2,6 +2,12 @@
   Impure default args for `pkgs/top-level/default.nix`. See that file
   for the meaning of each argument.
 */
+
+let
+
+  homeDir = builtins.getEnv "HOME";
+
+in

 {
   # We put legacy `system` into `localSystem`, if `localSystem` was not passed.
@@ -18,12 +24,25 @@

   # Fallback: The contents of the configuration file found at $NIXPKGS_CONFIG or
   # $HOME/.config/nixpkgs/config.nix.
-  config ? { },
+  config ?
+    let
+      configFile = builtins.getEnv "NIXPKGS_CONFIG";
+      configFile2 = homeDir + "/.config/nixpkgs/config.nix";
+      configFile3 = homeDir + "/.nixpkgs/config.nix"; # obsolete
+    in
+    if configFile != "" && builtins.pathExists configFile then
+      import configFile
+    else if homeDir != "" && builtins.pathExists configFile2 then
+      import configFile2
+    else if homeDir != "" && builtins.pathExists configFile3 then
+      import configFile3
+    else
+      { },

   # Overlays are used to extend Nixpkgs collection with additional
   # collections of packages.  These collection of packages are part of the
   # fix-point made by Nixpkgs.
-  overlays ? [ ],
+  overlays ? import ./impure-overlays.nix,

   crossOverlays ? [ ],

@@ -35,7 +54,7 @@
 assert args ? localSystem -> !(args ? system);
 assert args ? system -> !(args ? localSystem);

-import ./pure.nix (
+import ./. (
   removeAttrs args [ "system" ]
   // {
     inherit config overlays localSystem;
