diff --git a/pyproject.toml b/pyproject.toml
index 05cd96e..eb2000f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,8 +1,7 @@
 [build-system]
 build-backend = "setuptools.build_meta"
 requires = [
-  "setuptools>=77",
-  "setuptools-scm>=8",
+  "setuptools>=77",
 ]

 [project]
@@ -25,9 +24,7 @@ classifiers = [
   "Topic :: Software Development :: Libraries",
   "Topic :: Utilities",
 ]
-dynamic = [
-  "version",
-]
+version = "@version@"
 urls.Homepage = "https://github.com/pytest-dev/iniconfig"

 [dependency-groups]
@@ -41,9 +38,6 @@ packages = [ "iniconfig" ]
 [tool.setuptools.package-data]
 iniconfig = [ "py.typed" ]

-[tool.setuptools_scm]
-write_to = "src/iniconfig/_version.py"
-
 [tool.ruff]

 lint.extend-select = [
